training/model_management/src/azureml/model/mgmt/utils/common_utils.py

```diff
@@ -145,3 +146,4 @@ def tc_exception(e, message):
         tc.track_exception(value=e.__class__, properties={"exception": message})
     except Exception as e:
         print(f"Exception while logging exception to app insights: {e}")
+    tc.flush()
```

Comment:
have it within `try`? If tc is not properly initialized it would throw error

Reply:
Ack

Reply:
@skasturi without flush its not working for `custom_events`. In next iteration we would resolve this by using traces and letting appinsight handle flushing